vars/rollingUpgradePipeline.groovy

```diff
@@ -62,7 +62,7 @@ def call(Map pipelineParams) {
                     script {
                         def tasks = [:]
 
-                        for (version in supportedUpgradeFromVersions(env.GIT_BRANCH, pipelineParams.base_versions)) {
```

Comment:
can you revert this change ?

Reply:
we need this filter for OSS versions, so they won't upgrade from enterprise versions

Reply:
if i do, this test will not work, as the branch name does not include the format of enterprise branches (it is running from master branch). this test is intended to run from enterprise 2020.1 to latest enterprise master.
as @amoskong mentioned, we are much more strict with the verification of the base version vs new version.
so, we will need to improve it, by checking the tags of the image, if there are tags in it, otherwise, we will need to check it after we have a 1st node installed..

Reply:
for testing, you can do what you want.
this logic stays until we'll have better logic for the pipeline to select why list of version to start from.

I can't merge this, since it would start upgrades on OSS from 2020.1 -> master, or 2020.1 - > 4.4. 

both you and @amoskong  can thing of a better way todo that, but for now this logic would stays.

Reply:
i understand that, if someone tries to upgrade from enterprise to OSS it should fail, but, how many jobs/builds we have that this mechanism must stay to protect it?
the only time i saw this "failure" was when i was trying to run this, from a branch on my private fork that did not contain the enterprise and its version in the branch name...

> it would start upgrades on OSS from 2020.1 -> master, or 2020.1 - > 4.4

something or someone must trigger that, to have that "wrong" test running.. or our infra has autonomy to run rolling upgrades by itself?
i really want to understand the logic behind this mechanism today (in the past, when the procedures were less strict, maybe, it made sense, but now...)

Reply:
For now, we don't need tis change because i don't really care about enterprise branch.
We can have the job directly in 2020.1, we need to test upgrades from 2020.1.x to 2020.1.y.

Reply:
done.